azkaban-common/src/main/java/azkaban/utils/Utils.java

```diff
@@ -489,7 +490,22 @@ public static CronExpression parseCronExpression(String cronExpression, DateTime
     } else return null;
   }
 
-  public static boolean isCronExpressionValid(String cronExpression) {
-    return CronExpression.isValidExpression(cronExpression);
+  /**
+   *
+   * @param cronExpression
+   * @param timezone
+   * @return if the cronExpression is valid or not.
+   */
+  public static boolean isCronExpressionValid(String cronExpression, DateTimeZone timezone) {
+    if (!CronExpression.isValidExpression(cronExpression)) {
+      return false;
+    } else {
+      CronExpression cronExecutionTime = parseCronExpression(cronExpression, timezone);
+      if (cronExecutionTime == null || cronExecutionTime.getNextValidTimeAfter(new Date()) == null) {
```

Comment:
so we are trying to avoid scheduling flows in the past. right ? Can you please add a small note.

nit: else statement is not required.